kuma/search/serializers.py

```diff
@@ -6,13 +7,21 @@
 
 
 class SearchQuerySerializer(serializers.Serializer):
-    q = serializers.CharField(required=False)
+    q = serializers.CharField(
+        required=False, max_length=settings.ES_Q_MAXLENGTH)
     highlight = serializers.BooleanField(required=False, default=True)
     # Advanced search query paramenters.
     css_classnames = serializers.CharField(required=False)
     html_attributes = serializers.CharField(required=False)
     kumascript_macros = serializers.CharField(required=False)
 
+    def validate_q(self, value):
+        # Check that \n not in query
+        if r'\n' in value:
```

Comment:
```suggestion
        if '\n' in value:
```
that literal string business is only needed when doing those mock HTTP requests in `pytest`. Here it's just a regular string.

Reply:
If you just commit, I think tests should pass still.

Reply:
@peterbe when i remove the `r` in the validation it no longer throws an error when a `\n` is entered in the search box.

![Screen Shot 2019-12-03 at 12 00 33 PM](https://user-images.githubusercontent.com/1201062/70072215-8e190880-15c4-11ea-87f6-75dcbc5ab1c9.png)

Reply:
Weird. Perhaps it's a binary string. Either way. I'm happy to merge this!